flexget/options.py

```diff
@@ -473,3 +471,91 @@ def parse_args(self, *args, **kwargs):
         # Set the 'allow_manual' flag to True for any usage of the CLI
         setattr(result, 'allow_manual', True)
         return result
+
+
+class CLITableError(Exception):
+    """ A CLI table error"""
+
+
+class CLITable(object):
+    """
+    A data table suited for CLI output, created via its sent parameters.
+    """
+
+    def __init__(self, type, table_data, title=None, check_size=False):
+        self.title = title
+        self.type = type
+        self.check_size = check_size
+        self.table = self.supported_table_types()[type](table_data)
+
+    @property
+    def output(self):
+        self.table.title = self.title
+        if self.type == 'porcelain':
+            # porcelain is a special case of AsciiTable
+            self.table.inner_footing_row_border = False
+            self.table.inner_heading_row_border = False
+            self.table.outer_border = False
+        if not self.check_size or self.table.ok:
+            return '\n' + self.table.table
+        raise CLITableError(
+            'Terminal size is not suffice to display table. Terminal width is {}, table width is {}. '
+            'Consider setting a lower value using `--max-width` option.'.format(terminal_size()[0],
+                                                                                self.table.table_width))
+
+    @staticmethod
+    def supported_table_types(keys=False):
+        """
+        This method hold the dict for supported table type. Call with `keys=True` to get just the list of keys.
+        """
+        table_types = {
+            'plain': AsciiTable,
+            'porcelain': AsciiTable,
+            'single': SingleTable,
+            'double': DoubleTable,
+            'github': GithubFlavoredMarkdownTable
+        }
+        if keys:
+            return list(table_types)
+        return table_types
+
+    @staticmethod
+    def colorize(text, color_tag, porcelain=False):
```

Comment:
This is used outside from table context, in lines that have nothing to do with table. So this should probably be a plain function. Location is a bit icky though, coloring terminal output you need to import options.

Our console method is not much better being in logger.py, maybe both should be in "terminal.py" or so ...